### PR TITLE
[PROCEDURES] Explicit language on reversions and WIP

### PR DESCRIPTION
**Pull request by hexylena**
This commit implements #382 in response to the lack of explicit language
on how reversions are to be handled. This is in response to the events
transpiring in #433 and #449.

Additionally this PR corrects the misspelling of the pluralization of
the word veto.

| Key | Value |
| --- | --- |
| 192 hour window start | `2015-07-11 22:38:20+00:00` |
| 192 hour window end | `2015-07-19 22:38:20+00:00` |
| Current Committers | 19 |
| 25% quorum | `ceil(19 * .25)` = `ceil(4.75)` = 5, +1 votes |

cc @jmchilton @dannon @bgruening @jxtx 

(times via `TZ=UTC date --rfc-3339=seconds; TZ=UTC date --rfc-3339=seconds --date "192 hours"`)

### CODE REVIEW

**Comment by hexylena** [comment]
Just wanted to add a personal comment, I do not see the issues this PR is in response to as being anyone's fault, nor there being culpability to be tossed around here. 

I strongly believe this was simply an issue of lack of explicit language in procedures regarding merging PRs, aggravated by everyone being at conference/traveling. With this explicit language, we avoid similar situations in the future, though I'm not sure I have a solution for the "everyone is on conference" factor.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -163,6 +170,20 @@ For votes regarding non-coding issues such as procedure changes, the requirement
 that a veto is accompanied by a *technical* justification is relaxed somewhat,
 though a well reasoned justification must still be included.
 
+### Reversions
+
+A *-1* vote on any recently merged pull request is grounds for an immediate
```
Comment (jxtx) [inline CONTRIBUTING.md]: S/is grounds for/requires/ perhaps?

**Comment by jxtx** [comment]
+1

Also, I agree @erasche, this is just an instance of confusion where our policy was unclear, nobody was right or wrong. Thank you for working to refine the process.

**Comment by dannon** [comment]
+1, and, yes, I'll third that this was just confusion due to lack of clear policy, travel, etc, which is exactly why we needed this refinement.  Thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (hexylena) [inline CONTRIBUTING.md]: Good catch, that was my intention. 

I'll leave open for comments a while yet, then per procedures will open a new PR with that change included.

**Comment by bgruening** [comment]
:+1:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -163,6 +170,20 @@ For votes regarding non-coding issues such as procedure changes, the requirement
...
+reversion of the merged pull request. The backout of such a pull request
+invokes a mandatory, minimum 72 hour, review period.
+
+- Recently merged pull requests are defined as a being within the past 168 hours (7
+  days), so as to not prevent forward progress, while allowing for reversions of
+  things merged without proper review and consensus.
+- The person issuing the -1 vote may, upon commenting *-1* with technical
```
Comment (martenson) [inline CONTRIBUTING.md]: s/may/will perhaps? So we don't end up with inconsistent codebase?

File: CONTRIBUTING.md (hunk 1 above)
Comment (nsoranzo) [inline CONTRIBUTING.md]: @erasche Can you also fix https://github.com/galaxyproject/galaxy/pull/314#discussion_r31823679 ?

Comment (hexylena) [inline CONTRIBUTING.md]: Sure!

**Comment by natefoo** [comment]
+1

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -163,6 +170,20 @@ For votes regarding non-coding issues such as procedure changes, the requirement
...
+reversion of the merged pull request. The backout of such a pull request
+invokes a mandatory, minimum 72 hour, review period.
+
+- Recently merged pull requests are defined as a being within the past 168 hours (7
+  days), so as to not prevent forward progress, while allowing for reversions of
+  things merged without proper review and consensus.
+- The person issuing the -1 vote will, upon commenting *-1* with technical
+  justification per the vetoes section, immediately open and merge a pull request to
+  revert the original merge in question outside of the normal requirement for a
+  +1 from "someone other than the author", per "Handling Pull Requests"
```
Comment (jmchilton) [inline CONTRIBUTING.md]: I think given that a technical justification needs to be given - so I would suggest someone other than the -1 issuer needs to decide the justification is "technical" - not that they agree with it - just that the reason given is in fact technical.

"The person issuing the -1 vote will, upon commenting _-1_ with technical justification per the vetoes section, immediately open a pull request to revert the original merge in question. If any committer other than the -1 issuer deems the justification technical - regardless of whether they agree with justification - that committer must then merge the pull request to revert."

Comment (hexylena) [inline CONTRIBUTING.md]: Okay, I'll accept that premise. Will update wednesday when I'm back at work.

**Comment by hexylena** [comment]
@jmchilton implemented your language regarding technical justifications. Will probably re-open this, squashed, sometime tomorrow.

**Comment by martenson** [comment]
@erasche are you going to re-open this in a new PR?

**Comment by hexylena** [comment]
@martenson yes, per procedures specification I'll re-open as a squashed PR.

**Comment by hexylena** [comment]
Sorry, meant to do that and then took a nap instead.

**Pull request closed without merging**